### PR TITLE
Feature/pplus 1393 add a new check for unique names of modules

### PR DESCRIPTION
**Pull request by VladislavStrelchenko**
Ticket: https://spryker.atlassian.net/browse/PPLUS-1393

Change log: 
* Added check unique names of modules

### CODE REVIEW

File: src/Codebase/Infrastructure/SourceParser/StructureParser/ModuleParser.php [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+     * @param \Codebase\Application\Dto\SourceParserRequestDto $codebaseRequestDto
+     * @param \Codebase\Application\Dto\CodebaseSourceDto $codebaseSourceDto
+     *
+     * @return \Codebase\Application\Dto\CodebaseSourceDto
+     */
+    public function parse(SourceParserRequestDto $codebaseRequestDto, CodebaseSourceDto $codebaseSourceDto): CodebaseSourceDto
+    {
+        $coreModules = [];
+        foreach ($codebaseRequestDto->getCorePaths() as $corePath) {
+            foreach ((array)glob(sprintf(static::CORE_PATH_FORMAT, $corePath), GLOB_ONLYDIR) as $dir) {
```
Comment (kalaputsik) [inline src/Codebase/Infrastructure/SourceParser/StructureParser/ModuleParser.php:35]: could you move it to the variable like $directories or etc?
as far as It looks massive

File: src/Codebase/Infrastructure/SourceParser/StructureParser/ModuleParser.php [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+                $coreModules[] = $this->snakeCaseToCamelCase(basename((string)$dir));
+            }
+        }
+        $codebaseSourceDto->setCoreModuleNames(array_unique($coreModules, SORT_STRING));
+
+        $projectModules = [];
+        foreach ($codebaseRequestDto->getProjectPaths() as $projectPath) {
+            foreach ((array)glob(sprintf(static::PROJECT_PATH_FORMAT, $projectPath), GLOB_ONLYDIR) as $dir) {
```
Comment (kalaputsik) [inline src/Codebase/Infrastructure/SourceParser/StructureParser/ModuleParser.php:43]: could you move it to the method or at least to the variable?
as far as It looks massive

File: src/Codebase/Infrastructure/SourceParser/StructureParser/ModuleParser.php (hunk 1 above)
Comment (kalaputsik) [inline src/Codebase/Infrastructure/SourceParser/StructureParser/ModuleParser.php:35]: could we move it to the method or at least to the variable?
as far as It looks massive

File: src/Codebase/Infrastructure/SourceParser/StructureParser/ModuleParser.php (hunk 2 above)
Comment (kalaputsik) [inline src/Codebase/Infrastructure/SourceParser/StructureParser/ModuleParser.php:43]: could we move it to the method or at least to the variable?
as far as It looks massive

File: src/Upgrader/Configuration/ConfigurationProvider.php [hunk 3]
```diff
@@ -42,19 +42,18 @@ public function getIgnoreSources(): array
...
      */
     public function getCorePaths(): array
     {
-        $directories = [
-            $this->getVendorPath() . 'spryker' . DIRECTORY_SEPARATOR,
-            $this->getVendorPath() . 'spryker-eco' . DIRECTORY_SEPARATOR,
-            $this->getVendorPath() . 'spryker-shop' . DIRECTORY_SEPARATOR,
-        ];
+        $directories = [];
+        foreach ((array)glob($this->getVendorPath() . 'spryker*', GLOB_ONLYDIR) as $directory) {
```
Comment (kalaputsik) [inline src/Upgrader/Configuration/ConfigurationProvider.php:50]: it's not critical, but from my perspective construction with array_filter is easier to read

**Comment by kalaputsik** [comment]
Please fix the conflict with composer.lock

File: src/Codebase/Infrastructure/SourceParser/StructureParser/ModuleParser.php (hunk 2 above)
Comment (VladislavStrelchenko) [reply src/Codebase/Infrastructure/SourceParser/StructureParser/ModuleParser.php:43]: Thanks, extracted to an variable

File: src/Codebase/Infrastructure/SourceParser/StructureParser/ModuleParser.php (hunk 1 above)
Comment (VladislavStrelchenko) [reply src/Codebase/Infrastructure/SourceParser/StructureParser/ModuleParser.php:35]: Thanks, introduced $directories variable

File: src/Codebase/Infrastructure/SourceParser/StructureParser/ModuleParser.php (hunk 2 above)
Comment (VladislavStrelchenko) [reply src/Codebase/Infrastructure/SourceParser/StructureParser/ModuleParser.php:43]: Thanks, introduced $directories variable

File: src/Upgrader/Configuration/ConfigurationProvider.php (hunk 3 above)
Comment (VladislavStrelchenko) [reply src/Upgrader/Configuration/ConfigurationProvider.php:50]: Totally agree with you 
but in this case, we will have PhpStan issue(
Screenshot from 2022-07-15 13-10-36

Comment (VladislavStrelchenko) [reply src/Upgrader/Configuration/ConfigurationProvider.php]: Fixed

**Review by kalaputsik — APPROVED** [review]

**Review by DmytroKlymanSpryker — APPROVED** [review]

**Review by DmytroKlymanSpryker — APPROVED** [review]

**Pull request merged**